presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -649,7 +651,14 @@ private ConnectorTableMetadata getTableMetadata(Optional<Table> table, SchemaTab
             throw new PrestoException(HIVE_UNSUPPORTED_FORMAT, format("Not a Hive table '%s'", tableName));
         }
 
-        Function<HiveColumnHandle, ColumnMetadata> metadataGetter = columnMetadataGetter(table.get(), typeManager, metastoreContext.getColumnConverter());
+        List<TableConstraint<String>> tableConstraints = metastore.getTableConstraints(metastoreContext, tableName.getSchemaName(), tableName.getTableName());
+        List<String> notNullColumns = tableConstraints.stream()
+                .filter(NotNullConstraint.class::isInstance)
+                .map(constraint -> constraint.getColumns().stream()
+                        .findFirst()
+                        .orElseThrow(() -> new PrestoException(GENERIC_INTERNAL_ERROR, "NOT NULL constraint with no column")))
```

Comment:
Let's throw an external error here since the error is caused by invalid values we're getting from the metastore.  Also let's add the table name to the error message to help with debugging for queries that access multiple tables.